qiskit/result/distributions/probability.py

```diff
@@ -45,20 +45,18 @@ def __init__(self, data, shots=None):
             ValueError: If the string format of the keys is incorrect
         """
         self.shots = shots
+        self._num_bits = 0
         if data:
             first_key = next(iter(data.keys()))
             if isinstance(first_key, int):
-                pass
+                self._num_bits = len(bin(max(data.keys()))) - 2
```

Comment:
It would be good to add a comment explaining that this is not actually the number of qubits you measured, but just the number of bits needed to represent the largest key. It had me confused for a bit as I thought `num_bits` should be the number of qubits I measured! 😄

Reply:
done